packages/wagmi-connectors/src/createConnectorFromWallet.ts

```diff
@@ -0,0 +1,225 @@
+import type { AppMetadata, EIP1193Provider, Wallet } from '@mobile-wallet-protocol/client';
+import { ChainNotConfiguredError, type Connector, createConnector } from '@wagmi/core';
+import type { Omit } from '@wagmi/core/internal';
+import {
+  type AddEthereumChainParameter,
+  getAddress,
+  type Hex,
+  numberToHex,
+  type ProviderRpcError,
+  SwitchChainError,
+  UserRejectedRequestError,
+} from 'viem';
+
+import { toCamelCase } from './utils.js';
+
+type WagmiWallet = Wallet & {
+  wagmiType?: string;
+  supportsSimulation?: boolean;
+};
+
+export type CreateConnectorParameters = {
+  metadata: Omit<AppMetadata, 'appChainIds'>;
+  wallet: WagmiWallet;
+};
+
+export function createConnectorFromWallet(parameters: CreateConnectorParameters) {
+  type Provider = EIP1193Provider;
+
+  let walletProvider: Provider | undefined;
+
+  let accountsChanged: Connector['onAccountsChanged'] | undefined;
+  let chainChanged: Connector['onChainChanged'] | undefined;
+  let disconnect: Connector['onDisconnect'] | undefined;
+
+  const walletName = toCamelCase(parameters.metadata.appName);
+
+  return createConnector<Provider>((config) => ({
+    id: `${walletName}Mobile`,
+    name: `${walletName}Mobile`,
```

Comment:
Is there a reason why we're appending `Mobile`?

Reply:
originally was to avoid collision with existing coinbaseWallet connector, but think in the walletName we have suffix already, I'll remove it